evm/src/fixed_recursive_verifier.rs

```diff
@@ -819,7 +820,7 @@ where
         let has_not_parent_block = builder.sub(one, has_parent_block.target);
 
         // Check that the genesis block number is 0.
-        let gen_block_constr = builder.mul(has_not_parent_block, rhs.block_metadata.block_number);
+        let gen_block_constr = builder.mul(has_not_parent_block, lhs.block_metadata.block_number);
```

Comment:
This seems like something the test suite should have caught? I see the test changes below but I'm unsure if this change is covered now?

Reply:
The test suite didn't catch this because tests were all badly configured (with the first block having index 0 instead of 1), and wasn't caught when running the Ethereum test suite either because we don't use recursion there (too slow)